### PR TITLE
9th Terraform Recipe - Kafka with MongoDB source connector

### PR DESCRIPTION
**Pull request by dewan-ahmed**
# What changed, and why it matters

Resolves https://github.com/aiven/devportal/issues/871

### CODE REVIEW

File: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      end
+      subgraph Apache Kafka
+      Topic
+      end
+      id1 --->|polls changes| MongoDB-Source-Connector -->|publishes changes| Topic
+
+Describe the setup
+------------------
+
+Here is the Terraform recipe that will spin up an Aiven for Apache Kafka service with Kafka Connect enabled. This recipe will also create and configure a MongoDB source connector.
```
Comment (ftisiot) [inline docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:23]: Maybe we can add a link here to https://developer.aiven.io/docs/products/kafka/kafka-connect/howto/mongodb-poll-source-connector.html

File: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    plan                    = "business-4"
+    service_name            = join("-", [var.service_name_prefix, "kafka"])
+    maintenance_window_dow  = "sunday"
+    maintenance_window_time = "10:00:00"
+    kafka_user_config {
+      kafka_connect = true
+      kafka_rest    = true
+      kafka_version = "3.1"
+      kafka {
+        group_max_session_timeout_ms = 70000
```
Comment (ftisiot) [inline docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:46]: why do we have this custom settings? Are they strictly needed?

File: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      "output.format.key" : "json"
+      "publish.full.document.only" : "true"
+    }
+  }
+
+.. Tip:: 
+  
+  * Since you have ``kafka_connect`` set to ``true`` under the ``kafka_user_config``, you don't need a standalone Aiven for Apache Kafka Connect service.
+  * You need to enable the ``auto_create_topics_enable`` flag so that the connector is able to create the topic on the Apache Kafka cluster.
+  * The automatically created topic name will be **DATABASE.COLLECTION** - ``sample_airbnb.listingsAndReviews`` in this example.
```
Comment (ftisiot) [inline docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:75]:
```suggestion
  * The automatically created topic name will be  the concatenation of ``database`` and ``collection`` parameters - ``sample_airbnb.listingsAndReviews`` in this example.
```

**Review by ftisiot — CHANGES_REQUESTED** [review]
Good work, but I couldn't test. Do you have an existing MongoDB I can use for testing? Also left a couple comments

**Comment by ftisiot** [comment]
The code is working (I can see the mongodb data in Kafka) I'm ready to have another look at the surroundings once you fix the above issues!

File: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst (hunk 1 above)
Comment (dewan-ahmed) [reply docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:23]: Thanks. Good point. I'm adding the link when I first mention MongoDB source connector, i.e. in the first paragraph.

File: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst (hunk 2 above)
Comment (dewan-ahmed) [reply docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst]: They are not; removed. Good call.

**Comment by dewan-ahmed** [comment]
Thanks Francesco. I implemented the review feedback. Some notes:

1. I added MongDB connector link in the first paragraph where it's mentioned first.
2. I added a line that in case the DB is not reachable via public internet, it has to be part of a VPC. I think having a link so that the reader can learn more about VPC, would be helpful. But I couldn't find much info on DevPortal/Blog that I could link.

File: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      "output.format.value" : "json"
+      "output.format.key" : "json"
+      "publish.full.document.only" : "true"
+    }
+  }
+
+.. Tip:: 
+  
+  * Since you have ``kafka_connect`` set to ``true`` under the ``kafka_user_config``, you don't need a standalone Aiven for Apache Kafka Connect service.
+  * You need to enable the ``auto_create_topics_enable`` flag so that the connector is able to create the topic on the Apache Kafka cluster.
```
Comment (ftisiot) [inline docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst:72]:
```suggestion
  * The ``auto_create_topics_enable`` flag is enabled, therefore the connector is able to create the topic on the Apache Kafka cluster by pushing the first message, without having to create the topic first.
```

**Comment by dewan-ahmed** [comment]
Thanks Francesco. I added the Kafka Connect warning and reworded the topic auto creation sentence.

I removed the Tip from the explanation of the parameters since that section is not a tip. This also aligns with how other recipes are structured.

**Review by ftisiot — APPROVED** [review]
LGTM! Great work

**Pull request merged**